src/plugin/Manager.h

```diff
@@ -311,6 +311,18 @@ class Manager
 	 */
 	bool HookQueueEvent(Event* event) const;
 
+	/**
+	 * Hook that filters the dispatching of an event.
+	 *
+	 * @param event The event to be executed; it may be modified.
+	 *
+	 * @param no_remote True if the event should not be auto-published.
+	 *
+	 * @return Returns true if a plugin handled the dispatching; in that case
+	 * the plugin will have taken ownership.
```

Comment:
I think you modeled that comment on `HookQueueEvent`, right? I wonder if taking ownership is as clear here, or if it even applies. `EventMgr::Dispatch()` unrefs the event, so that happens regardless. In `EventDispatch()` there's [this block](https://github.com/zeek/zeek/blob/master/src/Event.cc#L64-L66) further down that manually unrefs the `obj` member — I think this means that for a plugin to do this right, it would have to know about this, which feels wrong ... we should probably do that unref if `PLUGIN_HOOK_WITH_RESULT` returns `true`. And the second part of this sentence could just go.

Thoughts?

Reply:
Gotcha, I was more or less blindly following `HookQueueEvent`. I think you're right: In contrast to the event queuing, this hook fires when the event *is already dispatched* from the perspective of the manager. We might prevent that it gets executed by altering how the event itself behaves but the dispatching cannot be undone. I added the `Unref(obj)` as suggested and revised the docstrings.